apps/aecore/src/aec_consensus_hc.erl

```diff
@@ -851,6 +1141,26 @@ get_entropy_hash(ChildEpoch) ->
             {error, not_in_cache}
     end.
 
+%% Final timeslot decisions, are cached so that we do not attempt to produce again when the decision
+%% has already been made. Example fact: We're not a leader and its not going to change in the same timeslot.
+-spec get_timeslot_skip_decision(Timeslot :: pos_integer()) -> false | atom().
+get_timeslot_skip_decision(Timeslot) ->
+    Decisions = aeu_ets_cache:lookup(?ETS_CACHE_TABLE, timeslot_skip_decision, #{}),
+    maps:get(Timeslot, Decisions, false).
```

Comment:
Generic question, do we really need all this caching of decisions, shouldn't/couldn't we just let the conductor loop sleep until a point in time where the decision is no longer relevant? There is a lot of complexity here that I'm not convinced is necessary...

Reply:
The conductor calls the consensus repeatedly with very small intervals, as there is no PoW. And it logs (debug logs) all the decision steps every time. Returning a status requesting a sleep (or sleeping in the consensus module) might work just as good. I will see to remove this and see what happens.